xbmc/guilib/GUIBaseContainer.h

```diff
@@ -152,8 +154,8 @@ class CGUIBaseContainer : public IGUIContainer
 
   int m_pageControl;
 
-  std::vector<CGUIListItemLayout> m_layouts;
-  std::vector<CGUIListItemLayout> m_focusedLayouts;
+  std::list<CGUIListItemLayout> m_layouts;
+  std::list<CGUIListItemLayout> m_focusedLayouts;
```

Comment:
these are looped over far more than they're resized so the cache locality of a vector most likely has better perf. your change optimizes for boot when the container needs to grow several times possibly.